### PR TITLE
Custom color profile file

### PR DESCRIPTION
**Pull request by luv4bytes**
Corrected the last one.

I added the first part of #14 .

It is possible to pass a color profile file as an argument now. I created a new subcommand for this, because the custom-colors subcommand uses positional arguments for the passed colors and I thought it might be a good idea to not mess around with this.

I also added an example of a color profile file in a new folder called examples and edited the README to include the new feature.

Please review and let me know if something should be done differently 😄

### CODE REVIEW

File: cherryrgb/src/lib.rs [hunk 1]
```diff
@@ -122,6 +117,41 @@ pub fn find_devices(product_id: Option<u16>) -> Result<Vec<(u16, u16)>> {
     Ok(usb_ids)
 }
 
+/// Reads the given color profile file and returns a vector of `ProfileKey`.
+/// # Arguments
+/// * `file_path` - Path of the file that holds the color profile information.
+pub fn read_color_profile_file(file_path: &str) -> Result<Vec<ProfileKey>> {
+    let mut f = std::fs::File::open(file_path).context(file_path.to_string())?;
```
Comment (skraus-dev) [inline cherryrgb/src/lib.rs:124]: Rename to `read_color_profile` and make it take a str-ref `color_profile: &str`.

Move the logic of opening the file and reading to string into the CLI main.rs.

This keeps the library function more universal - it shouldn't care where the input comes.

File: src/main.rs [hunk 2]
```diff
@@ -97,6 +103,18 @@ fn main() -> Result<()> {
 
             keyboard.set_custom_colors(keys)?;
         }
+        CliCommand::ColorProfileFile(args) => {
+            let colors_from_file = read_color_profile_file(&args.file_path)
```
Comment (skraus-dev) [inline src/main.rs:107]: As mentioned above, file open/read into string belongs here.

File: src/main.rs [hunk 3]
```diff
@@ -28,10 +29,16 @@ struct CustomColorOptions {
     colors: Vec<OwnRGB8>,
 }
 
+#[derive(StructOpt, Debug)]
+struct ColorProfileFileOptions {
+    file_path: String,
```
Comment (skraus-dev) [inline src/main.rs:34]: Change to

```rs
#[structopt(parse(from_os_str))]
file_path: PathBuf,
```

(`use std::path::PathBuf;`)

**Review by skraus-dev — CHANGES_REQUESTED** [review]
Just two little changes required, other than that it looks nice :+1:

**Comment by skraus-dev** [comment]
While at it, please also add a minimal test that deserializes a color profile with 2 or 3 entries.

**Comment by luv4bytes** [comment]
Made the changes and added a test for deserialization :)

File: cherryrgb/src/lib.rs [hunk 4]
```diff
@@ -576,4 +601,27 @@ mod tests {
...
+            }
+        "#;
+
+        let match_this: Vec<ProfileKey> = vec![
+            ProfileKey::new(0, OwnRGB8::new(255, 0, 0)),
+            ProfileKey::new(1, OwnRGB8::new(0, 255, 0)),
+            ProfileKey::new(2, OwnRGB8::new(0, 0, 255)),
+        ];
+
+        let profile_keys = read_color_profile(color_profile);
```
Comment (skraus-dev) [inline cherryrgb/src/lib.rs:621]: Change to

```rs
let profile_keys = read_color_profile(color_profile).expect("Failed deserializing color profile");
assert_eq!(match_this, keys);
```

**Review by skraus-dev — CHANGES_REQUESTED** [review]
Just a tiny one, then we are good to go

**Review by skraus-dev — APPROVED** [review]

**Pull request merged**